### ISSUE TITLE
Building CMake without git isn't working

### ISSUE THREAD

**Issue by iam-peter** [githubuser0xFFFF/Qt-Advanced-Docking-System#352]
When trying to build this project via CMake with the repository not being checked out from git (e.g. downloaded as an archive) it won't build due to CMake grabbing version information from git

```
string(REGEX REPLACE "^([0-9]+)\\..*" "\\1" VERSION_MAJOR "${GitTagVersion}")
string(REGEX REPLACE "^[0-9]+\\.([0-9]+).*" "\\1" VERSION_MINOR "${GitTagVersion}")
string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" VERSION_PATCH "${GitTagVersion}")
```
It will throw the following error:

```
CMake Error at CMakeLists.txt:9 (project):
  VERSION "HEAD-HASH-NOTFOUND.HEAD-HASH-NOTFOUND.HEAD-HASH-NOTFOUND" format
```

**Comment by githubuser0xFFFF** [comment]
Thank you peter for this report. The CMake build files are a contribution from an ADS user. At the moment you can use CMake build only, if you checkout via Git. I will add this information to the documentation,

**Comment by githubuser0xFFFF** [comment]
Fixed in master with pull request #384
